### DISCUSSION TITLE
Getting OBX Observation Value

### DISCUSSION THREAD

**Discussion by sergiopass01** [nHapiNET/nHapi#313] [General]
Hello

I am fairly new to HL7, so forgive me if I say something silly.  I have a requirement to accept any version of HL7 and, based on the message type pull specific data elements out of the message.  I have one requirement to get the observation value from an ORU^R01, specifically a base 64 encoded PDF in OBX5.1

Using some sample code, Is this the correct way of doing it?

```csharp
var data = oru.GetRESPONSE().GetORDER_OBSERVATION().GetOBSERVATION().OBX.GetObservationValue()[0].Data as NHapi.Model.V23.Datatype.ED;
Console.WriteLine(data.Data.Value);
```

Thank You!

**Comment by milkshakeuk** [comment]
Hi @sergiopass01 it looks okay to me.

I would do this, which is more or less the same as above except passing the repetition into `GetObservationValue`
```csharp
...
var parser = new PipeParser();
// you may need to set the default Obx2 type in case OBX-2 is empty and or the fall
// back incase what is provided at OBX-2 is not a valid type.
var options = new ParserOptions { DefaultObx2Type = "ED", InvalidObx2Type = "ED" };

var parsed = (ORU_R01)parser.Parse(message, options);
var data = parsed.GetRESPONSE().GetORDER_OBSERVATION().GetOBSERVATION().OBX.GetObservationValue(1).Data as ED;

Console.WriteLine(data.Data.Value);
```

FYI it might be slightly different for each message version, so that is correct for v23 but it might be slightly different for the other versions.

**Reply by sergiopass01** [reply]
thank you!
